homeassistant/components/todoist/calendar.py

```diff
@@ -419,33 +424,48 @@ def calendar_event(self) -> CalendarEvent | None:
         """Return the next upcoming calendar event."""
         if not self.event:
             return None
-        if not self.event.get(END) or self.event.get(ALL_DAY):
-            start = self.event[START].date()
-            return CalendarEvent(
-                summary=self.event[SUMMARY],
-                start=start,
-                end=start + timedelta(days=1),
-            )
+
+        end = self.event[END]
```

Comment:
What you have is fine. though do either of these also work to still keep this succint?
```
        start: datetime.datetime | datetime.date = self.event[START]
        end: datetime.datetime | datetime.date | None = self.event[END]
        if end is None or self.event.get(ALL_DAY):
            start = start.date()
            end = start + timedelta(days=1),
        return CalendarEvent(
            summary=self.event[SUMMARY], start=start, end=end
        )
```

or this?
```
        start: datetime.datetime = self.event[START]
        if self.event.get(ALL_DAY) or self.event[END] is None:
            return CalendarEvent(
                summary=self.event[SUMMARY], start=start.date(), end=start.date() + timedelta(days=1)
            )
        return CalendarEvent(
            summary=self.event[SUMMARY], start=start, end=self.event[END]
        )
```